provider/source/src/datetime/mod.rs

```diff
@@ -320,7 +320,7 @@ mod test {
 
         let cs_dates = DateLengths::from(&data);
```

Comment:
Thought: DateLengths and TimeLengths need to go. Maybe while you are in here you can get rid of them (can be in a follow-up).

Reply:
I don't know what it means to get rid of them; these are datetime datagen types.